### PR TITLE
Permitiendo al TA resolver clarificaciones de cursos

### PR DESCRIPTION
**Pull request by pabo99**
# Description

Se agrega la funcionalidad para que un Asistente de enseñanza (TA) pueda resolver
clarificaciones dentro de los cursos:

TeachingAssistantClarification

Fixes: #7445 

Fixes: #7453 

# Checklist:

- [x] The code follows the [coding guidelines](https://github.com/omegaup/omegaup/wiki/Coding-guidelines) of omegaUp.
- [x] The tests were executed and all of them passed.
- [x] If you are creating a feature, the new tests were added.
- [ ] If the change is large (> 200 lines), this PR was split into various Pull Requests. It's preferred to create one PR for changes in controllers + unit tests in PHPUnit,  and then another Pull Request for UI + tests in Jest, Cypress or both.

### CODE REVIEW

File: frontend/server/src/Authorization.php [hunk 1]
```diff
@@ -485,14 +485,25 @@ public static function isTeachingAssistant(
...
+        $identity,
+        $groups = []
     ): bool {
-        if (is_null($group) || is_null($identity->user_id)) {
+        if (empty($groups) || is_null($identity->user_id)) {
             return false;
         }
-        return self::isGroupMember($identity, $group);
+
+        if (self::isSystemAdmin($identity)) {
```
Comment (heduenas) [inline frontend/server/src/Authorization.php:500]: esto no deberia ir hatsa arriba de este funcion? O sea, que los system admins siempre sean TAs aun si $groups esta vacio.

**Review by heduenas — APPROVED** [review]

File: frontend/server/src/Authorization.php (hunk 1 above)
Comment (pabo99) [reply frontend/server/src/Authorization.php:496]: Es correcta tu observación. Invertiré el orden de las condiciones

File: frontend/server/src/DAO/Courses.php [hunk 2]
```diff
@@ -1608,4 +1608,39 @@ public static function getActivityReport(
             'totalRows' => $totalRows,
         ];
     }
+    /**
+     * @return list<\OmegaUp\DAO\VO\Groups>
+     */
+    public static function getCourseTeachingAssistantsGroups(
```
Comment (heduenas) [inline frontend/server/src/DAO/Courses.php:1614]: Renombra esta funcion a `getCourseTeachingAssistantGroup`

Comment (heduenas) [reply frontend/server/src/DAO/Courses.php:1614]: Mas bien a `getCourseTeachingAssistantGroups`

**Review by heduenas — COMMENTED** [review]
Podrias agregar un test donde un TA intente resolver una clarificacion?

File: frontend/server/src/DAO/Courses.php (hunk 2 above)
Comment (pabo99) [reply frontend/server/src/DAO/Courses.php]: Listo

**Comment by pabo99** [comment]
> Podrias agregar un test donde un TA intente resolver una clarificacion?

Listo

**Review by heduenas — APPROVED** [review]

**Pull request merged**